iosHackthonApp/Modules/Features/Login/Sources/Login/LoginViewModel.swift

```diff
@@ -10,37 +10,64 @@ import SwiftUI
 import shared
 import Components
 import Theming
+import Strings
 
 public final class LoginViewModel: ObservableObject {
 
-	@Published var email: String = ""
-	@Published var password: String = ""
-	@Published var showingAlert = false
-	@Published var isLoading: Bool = false
-	@Published var coloredNavAppearance = UINavigationBarAppearance()
-	@Published var firebase: FirebaseAuthenticationStore
-	@ObservedObject var authorizationStore: AuthorizationStore
-
-	public init(firebase: FirebaseAuthenticationStore, authorizationStore: AuthorizationStore) {
-		self.firebase = firebase
-		self.authorizationStore = authorizationStore
-		coloredNavAppearance.configureWithOpaqueBackground()
-		coloredNavAppearance.backgroundColor = UIColor(ColorManager.appPrimary)
-		coloredNavAppearance.titleTextAttributes = [.foregroundColor: UIColor.white]
-		coloredNavAppearance.largeTitleTextAttributes = [.foregroundColor: UIColor.white]
-	}
-
-	func login(email: String, password: String) {
-		//TODO: API key to be handled in shared layer. remove the apikey parameter once implemented
-		// To run the project for now, add the api key and run
-		isLoading = true
-		firebase.signIn(apiKey: "", email: email, password: password, returnSecureToken: true, completionHandler: { result, error in
-			if (result?.idToken != nil) {
-				self.authorizationStore.storeUserLoggedInStatus(true)
-			} else {
-				self.showingAlert.toggle()
-			}
-			self.isLoading = false
-		})
-	}
+    typealias strings = Strings.Login
+
+    @Published var firebase: FirebaseAuthenticationStore
+    @ObservedObject var authorizationStore: AuthorizationStore
+
+    public init(firebase: FirebaseAuthenticationStore, authorizationStore: AuthorizationStore) {
+        self.firebase = firebase
+        self.authorizationStore = authorizationStore
+        coloredNavAppearance.configureWithOpaqueBackground()
+        coloredNavAppearance.backgroundColor = UIColor(ColorManager.appPrimary)
+        coloredNavAppearance.titleTextAttributes = [.foregroundColor: UIColor.white]
+        coloredNavAppearance.largeTitleTextAttributes = [.foregroundColor: UIColor.white]
+    }
+
+    @Published var email: String = ""
+    @Published var password: String = ""
+    @Published var showingAlert = false
+    @Published var isLoading: Bool = false
+    @Published var coloredNavAppearance = UINavigationBarAppearance()
+
+    let title = strings.heading
```

Comment:
I think it's OK to do not have inside a ViewModel theses properties. For me, the ViewModel should reflect a state. In this case, we should know the state of email, password, showingAlert or isLoading which is correct. But for static texts, the right place is inside the view.

Reply:
I don't know, the last few projects I worked on we would create a viewModel even if it was **just** to hold on to static texts.
I can change it back though if everyone wants - I don't feel particularly strongly about it either way